### PR TITLE
fix(config): disallow URLs ending with `/ghost`

### LINKED ISSUE #754: Ghost setup should fail for /ghost subdirs
Although I can't find the documentation for it, you can't set your Ghost Installation URL with a trailing `/ghost` (i.e. https://your.blog/ghost [https://your.blog/ghost/ghost would be the admin url]). If you do and you try to access the admin interface, you're going to have a bad time (because infinite redirects).

Since Ghost doesn't support this, the CLI should fail when the user tries to set their url to end with `/ghost`

Also, the admin URL doesn't work with this because of the way the regex was designed - you would get 
```
You can access your publication at https://your.blog/ghost
Your admin interface is located at https://your.blog/ghost/
```
This doesn't need to be fixed, but I found it interesting

### PR DESCRIPTION
**Pull request by vikaspotluri123**
closes #754 

I'm not sure if there should be an additional check w/ `ghost start` (it's possible to manually configure the URL so the config check wouldn't be run) - adding it would require additional checks because the admin URL could differ from the hosted URL

### CODE REVIEW

File: lib/commands/config/advanced.js [hunk 1]
```diff
@@ -21,8 +21,15 @@ const knownMailServices = [
 module.exports = {
     url: {
         description: 'Blog URL with protocol E.g. http://loveghost.com',
-        validate: value => validator.isURL(value, {require_protocol: true}) ||
-            'Invalid URL. Your URL should include a protocol, E.g. http://my-ghost-blog.com',
+        validate: (value) => {
+            const urlValidation = validator.isURL(value, {require_protocol: true});
+            const endsWithGhost = new RegExp('(?:/ghost/?)$', 'i');
```
[the remark this answers is not in the document]
Comment (vikaspotluri123) [reply lib/commands/config/advanced.js]: Yeah, I'm not sure why I put a non-capturing capturing group 🤦‍♂️

**Comment by vikaspotluri123** [comment]
To DRY it up, would we want to put the URL validator in the `utils` folder, or can we leave it in the config folder? I'm thinking utils would be better...

**Comment by vikaspotluri123** [comment]
I've updated the branch to how I think it should be done 😄

**Comment by acburdine** [comment]
Gonna go ahead and merge this - we can change the wording later if it's confusing 😄

**Pull request merged**